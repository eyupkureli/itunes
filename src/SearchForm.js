import React from 'react';
import styled from 'styled-components';
export const SearchFormRow = styled.div`
    padding: 10px;
    background-color: #c4edf7;
`;

export const Label = styled.label`
    width: 120px;
    display: inline-block;
`;

function SearchForm(props) {

    return (
        <>
            <SearchFormRow>
                <Label>
                    Search Type : 
                </Label>
                <select value={props.searchType} onChange={props.onSearchTypeChange}>
                    <option value="musicArtist">Music Artist</option>
                    <option value="album">Album</option>
                    <option value="song">Song</option>
                </select>
            </SearchFormRow>
            <SearchFormRow>
                <Label>
                    Search Text :
                </Label>
                <input type="text" value={props.searchText} onChange={props.onSearchTextChange} />
            </SearchFormRow>
        </>
    );
}

export default SearchForm;