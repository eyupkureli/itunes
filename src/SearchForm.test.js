import React from 'react';
import { render, waitForElementToBeRemoved, fireEvent, within } from '@testing-library/react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import SearchForm from './SearchForm';

describe('SearchForm', () => {

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<SearchForm debug />);
        expect(component).toMatchSnapshot();
    });

    it('renders', () => {
        const wrapper = shallow(<SearchForm />);
        expect(wrapper.exists()).toBe(true);
    });

    it('should display searchText', async () => {
        const onSearchTypeChange = jest.fn();
        const onSearchTextChange = jest.fn();

        const container = render(<SearchForm searchText="erol" searchType="album" onSearchTypeChange={onSearchTypeChange} onSearchTextChange={onSearchTextChange} />);
        expect(container.getByDisplayValue('erol')).toBeInTheDocument();
    });

});


