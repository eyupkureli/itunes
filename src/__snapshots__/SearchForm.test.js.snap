// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should get searchText property 1`] = `ShallowWrapper {}`;

exports[`SearchForm should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
